src/utils/errorSerializer.ts

```diff
@@ -34,61 +38,64 @@ import UserTimeoutError from "../errors/UserTimeoutError";
 import VersionInvalidError from "../errors/VersionInvalidError";
 import VersionMismatchError from "../errors/VersionMismatchError";
 
-const errorCodeToErrorClass: {[key: string]: any} = {
+const errorCodeToErrorClass: Record<keyof typeof ErrorCode, KnownErrorConstructors> = {
   [ErrorCode.ERR_WEBEID_ACTION_PENDING]:          ActionPendingError,
   [ErrorCode.ERR_WEBEID_ACTION_TIMEOUT]:          ActionTimeoutError,
   [ErrorCode.ERR_WEBEID_CONTEXT_INSECURE]:        ContextInsecureError,
   [ErrorCode.ERR_WEBEID_EXTENSION_UNAVAILABLE]:   ExtensionUnavailableError,
+  [ErrorCode.ERR_WEBEID_MISSING_PARAMETER]:       MissingParameterError,
   [ErrorCode.ERR_WEBEID_NATIVE_INVALID_ARGUMENT]: NativeInvalidArgumentError,
   [ErrorCode.ERR_WEBEID_NATIVE_FATAL]:            NativeFatalError,
   [ErrorCode.ERR_WEBEID_NATIVE_UNAVAILABLE]:      NativeUnavailableError,
   [ErrorCode.ERR_WEBEID_USER_CANCELLED]:          UserCancelledError,
   [ErrorCode.ERR_WEBEID_USER_TIMEOUT]:            UserTimeoutError,
   [ErrorCode.ERR_WEBEID_VERSION_INVALID]:         VersionInvalidError,
   [ErrorCode.ERR_WEBEID_VERSION_MISMATCH]:        VersionMismatchError,
+  [ErrorCode.ERR_WEBEID_UNKNOWN_ERROR]:           UnknownError,
 };
 
-export function serializeError(error: any): any {
-  const {
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  } = error;
+export function serializeError(error: Error): SerializedError {
+  if (typeof error != "object" || error == null || !(error instanceof Error)) {
+    throw new UnknownError();
+  }
+
+  const result: Record<string, unknown> = {};
+
+  for (const prop of Object.getOwnPropertyNames(error)) {
+    Reflect.set(result, prop, Reflect.get(error, prop));
+  }
+
+  return result as SerializedError;
+}
 
-  return {
-    ...(
-      Object.fromEntries(
-        Object.getOwnPropertyNames(error)
-          .map((prop) => [prop, error[prop]])
-      )
-    ),
+function isSerializedError(obj: unknown): obj is SerializedError {
+  if (typeof obj !== "object" || obj === null) {
+    return false;
+  }
 
-    message,
-    name,
-    fileName,
-    lineNumber,
-    columnNumber,
-    stack,
-  };
+  const errorObject = obj as SerializedError;
+
+  return (
+    typeof errorObject.name    === "string" &&
+    typeof errorObject.message === "string" &&
+    typeof errorObject.stack   === "string"
+  );
 }
 
-export function deserializeError(errorObject: any): any {
+export function deserializeError(errorObject: SerializedError): InstanceType<KnownErrorConstructors> {
   let error;
 
-  if (typeof errorObject.code == "string" && errorObject.code in errorCodeToErrorClass) {
+  if (isSerializedError(errorObject)) {
     const CustomError = errorCodeToErrorClass[errorObject.code];
 
     error = new CustomError();
```

Comment:
Just checking, can every constructor in `errorCodeToErrorClass`  be called with no arguments?

Reply:
Before this PR you couldn't construct some errors without arguments, but those were changed (MissingParameterError and VersionMismatchError) - Making the TypeScript linter rules stricter revealed that.